spec/features/admin_views_councillor_contribution_spec.rb

```diff
@@ -44,17 +44,68 @@
     expect(page).to have_content "Felix Chaung #{Time.current.strftime('%B %d, %Y %H:%M')} Casey City Council"
   end
 
-  it "successfully with suggested councillors on the show page" do
+  it "successfully with suggested councillors on the show page with contributor information" do
     sign_in_as_admin
 
     click_link "Councillor Contributions"
 
     click_link "View"
 
-    expect(page).to have_content "Contributor Felix Chaung"
-    expect(page).to have_content "Authority Casey City Council"
+    expect(page).to have_content "Felix Chaung"
+    expect(page).to have_content "example@gmail.com"
+    expect(page).to have_content "Casey City Council"
     expect(page).to have_content "Mila Gilic mgilic@casey.vic.gov.au"
     expect(page).to have_content "Susan Serey sserey@casey.vic.gov.au"
     expect(page).to have_content "Rosalie Crestani rcrestani@casey.vic.gov.au"
   end
 end
+
+  feature "Admin views councillor contributions without contributor information" do
+    before :each do
+        Timecop.freeze(Time.local(2017, 8, 1))
+
+        authority = create(:authority, full_name: "Casey City Council")
+        councillor_contribution = create(:councillor_contribution, contributor: nil, authority: authority)
+        creation_time = Time.current
+
+        create(
+          :suggested_councillor,
+          name: "Mila Gilic",
+          email: "mgilic@casey.vic.gov.au",
+          councillor_contribution: councillor_contribution,
+          created_at: creation_time
+        )
+        create(
+          :suggested_councillor,
+          name: "Susan Serey",
+          email: "sserey@casey.vic.gov.au",
+          councillor_contribution: councillor_contribution,
+          created_at: creation_time
+        )
+        create(
+          :suggested_councillor,
+          name: "Rosalie Crestani",
+          email: "rcrestani@casey.vic.gov.au",
+          councillor_contribution: councillor_contribution,
+          created_at: creation_time
+        )
+    end
+
+    after :each do
+      Timecop.return
+    end
+
+    it "successfully" do
+      sign_in_as_admin
+
+      click_link "Councillor Contributions"
+
+      click_link "View"
+
+      expect(page).to have_content "Anonymous Contributor"
+      expect(page).to have_content "Casey City Council"
```

Comment:
Does this test need to cover all this other information? It reads to me like the purpose of this test is to make sure it says 'Anonymous contributor' when the contributor hasn't added their details?

This points to a good potential refactor here to me. The logic of how to display the name of the person is currently in the view, but it should be extracted to the model or a view helper—I think the model probably. Then we could also just cover the logic in a unit test. Could be a good pairing session?